### PR TITLE
kingfisher process api: convert file item data to str before sending …

### LINKED ISSUE #608: JSONDecodeError: Expecting value: line 1 column 1 (char 0)
Sentry Issue: [KINGFISHER-PROCESS-7Y](https://sentry.io/organizations/open-contracting-partnership/issues/2219589528/?referrer=github_integration)

```
JSONDecodeError: Expecting value: line 1 column 1 (char 0)
(6 additional frame(s) were not displayed)
...
  File "flask/views.py", line 89, in view
    return self.dispatch_request(*args, **kwargs)
  File "ocdskingfisherprocess/web/views_api_v1.py", line 160, in dispatch_request
    data = json.loads(request.form.get('data'))
  File "__init__.py", line 354, in loads
    return _default_decoder.decode(s)
  File "json/decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "json/decoder.py", line 357, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
```

### PR DESCRIPTION
**Pull request by yolile**
…it to process

closes #608

### CODE REVIEW

File: kingfisher_scrapy/extensions.py [hunk 1]
```diff
@@ -199,6 +199,10 @@ def item_scraped(self, item, spider):
         if isinstance(item, FileItem):
             data['number'] = item['number']
             data['data'] = item['data']
```
Comment (jpmckinney) [inline kingfisher_scrapy/extensions.py:201]: Delete this line.

**Review by jpmckinney — DISMISSED** [review]

**Comment by jpmckinney** [comment]
Actually, can you look at https://github.com/open-contracting/kingfisher-collect/issues/371? The data can sometimes be bytes, instead of str.

**Comment by yolile** [comment]
@jpmckinney hmm that issue is about Digiwhist but now Digiwhist pass through the line delimited middleware as well. But the last comment in the issue also mentions that bytes is preferred over str, so do you want me to change it to bytes instead?

**Comment by jpmckinney** [comment]
I closed #371 as I don't think it's a general problem.

However, it led me to review how we use encode/decode/json.dump/json/load/bytes/str and I noticed a few things:

* I see one case in `mexico_quien_es_quien` where JSON data is dumped then encoded to bytes, which I don't think is necessary. And, actually, I think that spider could just use `root_path` instead of indexing to the `'data'` key.

Otherwise, most of the relevant code is in `middlewares.py`:

* ResizePackageMiddleware dumps the package, but I don't think this is necessary (it's done by the extensions). However, we'd need to deepcopy the package metadata if we remove that line.
* AddPackageMiddleware loads the data to add `{"releases":[...]}` around the data. Technically, we could just do this using an f-string if the input is a str or bytes.
* LineDelimitedMiddleware decodes the bytes to str, but I don't remember why this behavior was introduced in https://github.com/open-contracting/kingfisher-collect/commit/7d406748a36cf073f2cff93a80c23c324324d6f2

File: kingfisher_scrapy/extensions.py [hunk 2]
```diff
@@ -198,7 +198,10 @@ def item_scraped(self, item, spider):
 
         if isinstance(item, FileItem):
             data['number'] = item['number']
-            data['data'] = item['data']
+            if isinstance(item['data'], str):
```
Comment (jpmckinney) [inline kingfisher_scrapy/extensions.py:201]: Would this work, too?

```suggestion
            if isinstance(item['data'], (bytes, str)):
```

**Review by jpmckinney — COMMENTED** [review]
I left one review comment. We can create a separate issue for my other comment if we want to address those separately.

**Comment by yolile** [comment]
@jpmckinney I was trying to at least resolve the mexico_quien_es_quien issue but it seems like it will require more changes than expected so it is better to just move your comments to another issue (#616 )

**Review by jpmckinney — APPROVED** [review]

**Pull request merged**